### PR TITLE
Prioritized updates

### PR DESCRIPTION
**Pull request by niik**
xref https://github.com/github/desktop/issues/894 

## Description

From time to time we ship new releases containing fixes for high severity security vulnerabilities. Most frequently this has been a vulnerability in a third-party dependency (Git, Git LFS etc) but it could also be a vulnerability within Desktop itself. Over Desktop's [lifetime](https://github.blog/news-insights/announcing-github-for-mac/) we've had a few releases where we've wanted to get the word out to users that they should update as soon as possible to stay safe.

With this change we'll be able to communicate to Desktop clients that they should prioritize updating to the latest version ASAP. When we do so we'll make remove the ability to dismiss the update banner, as well as change the styling and text to convey the urgency of updating.

### Screenshots

A screenshot of GitHub Desktop using the dark theme showing a banner with the text This version of GitHub Desktop is missing important updates. Please restart GitHub Desktop now to install pending updates

A screenshot of GitHub Desktop using the light theme showing a banner with the text This version of GitHub Desktop is missing important updates. Please restart GitHub Desktop now to install pending updates

## Release notes

Notes:

### CODE REVIEW

File: app/styles/themes/_dark.scss [hunk 1]
```diff
@@ -376,6 +376,11 @@ body.theme-dark {
   --dialog-information-color: #{$blue-400};
   --dialog-error-color: #{$red-600};
 
+  /** Banner */
+  --banner-warning-background: #{darken($orange-900, 20%)};
+  --banner-warning-text-color: var(--text-color);
+  --banner-warning-link-color: #4285fb;
```
Comment (tidy-dev) [inline app/styles/themes/_dark.scss]: Good contrast for dark mode. :)

File: app/styles/_variables.scss [hunk 2]
```diff
@@ -477,6 +477,11 @@ $overlay-background-color: rgba(0, 0, 0, 0.4);
   --dialog-information-color: #{$blue-400};
   --dialog-error-color: #{$red};
 
+  /** Banner */
+  --banner-warning-background: #{$yellow-700};
+  --banner-warning-text-color: var(--text-color);
+  --banner-warning-link-color: #{$blue-700};
```
Comment (tidy-dev) [inline app/styles/_variables.scss]: For accessibility, we need:
link -> text = 3:1 
link/text -> background: 4.5:1 

Unfortunately, this is giving link:text of 1.5:1 for light mode. 

Here is a link checker tool that has inputs for the link, the text, and the background to make verification easier.
https://webaim.org/resources/linkcontrastchecker/

Feel free to fix as you see fit. When I was testing locally, I tried the hidden bidi characters styles (which have been modified recently for accessibility.)  These colors were inspired by warning colors of https://primer.style/components/banner. ... which one could argue that maybe we could use there "Critical" banner styles which would be a redish color kind of like dark mode. https://primer.style/foundations/primitives/color has a `--bgColor-danger-muted` (for background) and `--bgColor-danger-emphasis` (for icon) -- but would need to be verified for accessibility. (Notes, the icon needs to have 3:1 to background)

```
  /** Banner */
  --banner-warning-background: #{$yellow-100}; // copied from --file-warning-background-color
  --banner-warning-text-color: var(--text-color);
  --banner-warning-link-color: var(--link-button-color);
  
  /_update-available.css 
    &.priority {
      .octicon {
          color: var(--file-warning-color);
        }
    ...
  ```
Showing using same styles as from bidi character warning

**Review by tidy-dev — CHANGES_REQUESTED** [review]
Looking good! 

I have left a comment as we have one accessibility contrast issue to resolve, but other than that looks great.

**Comment by niik** [comment]
> I have left a comment as we have one accessibility contrast issue to resolve, but other than that looks great.

Thanks a bunch. I aligned the dark banner with the file warning in the dark theme and (believe) I've covered all the contrast issues

A screenshot of GitHub Desktop using the dark theme showing a banner with the text This version of GitHub Desktop is missing important updates. Please restart GitHub Desktop now to install pending updates
A screenshot of GitHub Desktop using the light theme showing a banner with the text This version of GitHub Desktop is missing important updates. Please restart GitHub Desktop now to install pending updates

**Review by tidy-dev — CHANGES_REQUESTED** [review]
Super close... :)

In light and dark theme we are not quite to the 3:1 for link to body text. 
 
CleanShot 2024-12-05 at 09 00 22

CleanShot 2024-12-05 at 09 05 23


I found if we used the original link button blue it works on both dark/light themes, so I think we can just not apply any special css to the links in this banner and we will be good. 

CleanShot 2024-12-05 at 09 02 06

CleanShot 2024-12-05 at 09 05 47

**Comment by tidy-dev** [comment]
Lol.. after I posted I didn't catch that the link to background failing now.

image

So, we need to figure that out... I wonder if bidi character is failing that.

**Comment by niik** [comment]
Alright, I somehow missed the contrast between the text and the link and only cared about link to background. Here's [new results for the light theme](https://webaim.org/resources/linkcontrastchecker/?fcolor=24292E&bcolor=FFFBDD&lcolor=046EE7) and [dark theme](https://webaim.org/resources/linkcontrastchecker/?fcolor=F6F8FA&bcolor=272216&lcolor=6682FF)

image

image

**Review by tidy-dev — APPROVED** [review]
💖  Thank you for those fixes!

**Comment by 72158895omar1** [comment]
> xref https://github.com/github/desktop/issues/894
> 
> ## Description
> From time to time we ship new releases containing fixes for high severity security vulnerabilities. Most frequently this has been a vulnerability in a third-party dependency (Git, Git LFS etc) but it could also be a vulnerability within Desktop itself. Over Desktop's [lifetime](https://github.blog/news-insights/announcing-github-for-mac/) we've had a few releases where we've wanted to get the word out to users that they should update as soon as possible to stay safe.
> 
> With this change we'll be able to communicate to Desktop clients that they should prioritize updating to the latest version ASAP. When we do so we'll make remove the ability to dismiss the update banner, as well as change the styling and text to convey the urgency of updating.
> 
> ### Screenshots
> A screenshot of GitHub Desktop using the dark theme showing a banner with the text This version of GitHub Desktop is missing important updates. Please restart GitHub Desktop now to install pending updates
> 
> A screenshot of GitHub Desktop using the light theme showing a banner with the text This version of GitHub Desktop is missing important updates. Please restart GitHub Desktop now to install pending updates
> 
> ## Release notes Omar a Rodríguez botja
> Notes:

72158895omat1

File: app/styles/themes/_dark.scss (hunk 1 above)
Comment (dav3yblaz3) [reply app/styles/themes/_dark.scss]: thank you yes couldnt agree more

**Comment by F-fengzi** [comment]
That did not take too long to put into use! :D

File: app/styles/themes/_dark.scss (hunk 1 above)
Comment (137613413) [reply app/styles/themes/_dark.scss]: Name (required

File: app/styles/_variables.scss (hunk 2 above)
Comment (skyskygreen93) [reply app/styles/_variables.scss]: /** Banner */
  --banner-warning-background: #{$yellow-100}; // copied from --file-warning-background-color
  --banner-warning-text-color: var(--text-color);
  --banner-warning-link-color: var(--link-button-color);
  
  /_update-available.css 
    &.priority {
      .octicon {
          color: var(--file-warning-color);
        }
    ...

**Pull request merged**